concordium-node/src/grpc2.rs

```diff
@@ -1367,6 +1370,374 @@ pub mod server {
+                        }
+                    })
+                    .collect();
+                Ok(tonic::Response::new(crate::grpc2::types::BannedPeers {
+                    peers,
+                }))
+            } else {
+                Err(tonic::Status::internal("Could not load banned peers."))
+            }
+        }
+
+        async fn ban_peer(
+            &self,
+            request: tonic::Request<crate::grpc2::types::PeerToBan>,
+        ) -> Result<tonic::Response<crate::grpc2::types::Empty>, tonic::Status> {
+            let ip = request.into_inner().ip_address.require()?;
+            match <std::net::IpAddr as std::str::FromStr>::from_str(&ip.value) {
+                Ok(ip_addr) => match self.node.drop_by_ip_and_ban(ip_addr) {
+                    Ok(_) => Ok(tonic::Response::new(crate::grpc2::types::Empty {})),
+                    Err(e) => Err(tonic::Status::internal(format!("Could not ban peer {}.", e))),
+                },
+                Err(e) => Err(tonic::Status::invalid_argument(format!(
+                    "Invalid IP address provided {}",
+                    e
+                ))),
+            }
+        }
+
+        async fn unban_peer(
+            &self,
+            request: tonic::Request<crate::grpc2::types::BannedPeer>,
+        ) -> Result<tonic::Response<crate::grpc2::types::Empty>, tonic::Status> {
+            match <std::net::IpAddr as std::str::FromStr>::from_str(
+                &request.into_inner().ip_address.require()?.value,
+            ) {
+                Ok(ip_addr) => {
+                    let banned_id = crate::p2p::bans::PersistedBanId::Ip(ip_addr);
+                    match self.node.unban_node(banned_id) {
+                        Ok(_) => Ok(tonic::Response::new(crate::grpc2::types::Empty {})),
+                        Err(e) => {
+                            Err(tonic::Status::internal(format!("Could not unban peer {}.", e)))
+                        }
+                    }
+                }
+                Err(e) => {
+                    Err(tonic::Status::invalid_argument(format!("Invalid IP address {}.", e)))
+                }
+            }
+        }
+
+        #[cfg(feature = "network_dump")]
+        async fn dump_start(
+            &self,
+            request: tonic::Request<crate::grpc2::types::DumpRequest>,
+        ) -> Result<tonic::Response<crate::grpc2::types::Empty>, tonic::Status> {
+            let file_path = request.get_ref().file.to_owned();
+            // todo: fail with invalid argument if file path was empty.
+            let result = self
+                .node
+                .activate_dump(
+                    if file_path.is_empty() {
+                        "dump"
+                    } else {
+                        &file_path
+                    },
+                    request.get_ref().raw,
+                )
+                .is_ok();
+            Ok(tonic::Response::new(crate::grpc2::types::Empty {}))
```

Comment:
Would it make sense to be more defensive here and only allow relative paths inside, e.g., a data directory?

Reply:
I'm not sure. We assume the operator has exclusive access to his node anyhow - why not let him dump the file where ever he wants to? 
On the other hand it also sounds fine to somehow sanitize the path. 

I'll let that decision be up to the reviewer :) ...

Reply:
I guess as long as it doesn't crash the node it should be fine. So let's keep it simple.